airbyte-integrations/connectors/source-oracle-v2/src/cdk/main/kotlin/io/airbyte/cdk/command/ConnectorCommandLinePropertySource.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.command
+
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.cdk.operation.CONNECTOR_OPERATION
+import io.airbyte.cdk.operation.OperationType
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.env.MapPropertySource
+import io.micronaut.core.cli.CommandLine
+import java.io.File
+import java.nio.file.Path
+
+private val log = KotlinLogging.logger {}
+
+/**
+ * Custom Micronaut [MapPropertySource] that reads the command line arguments provided via the
+ * connector CLI and turns them into configuration properties. This allows the arguments to be
+ * injected into code that depends on them via Micronaut.
+ */
+class ConnectorCommandLinePropertySource(commandLine: CommandLine) :
+    MapPropertySource("connector", resolveValues(commandLine))
+
+const val CONNECTOR_CONFIG_PREFIX: String = "airbyte.connector.config"
+const val CONNECTOR_CATALOG_PREFIX: String = "airbyte.connector.catalog"
+const val CONNECTOR_STATE_PREFIX: String = "airbyte.connector.state"
+const val CONNECTOR_OUTPUT_FILE = "airbyte.connector.output.file"
+
+private fun resolveValues(commandLine: CommandLine): Map<String, Any> {
+    val ops: List<OperationType> =
+        OperationType.entries.filter { commandLine.optionValue(it.name.lowercase()) != null }
+    if (ops.isEmpty()) {
+        throw IllegalArgumentException("Command line is missing an operation.")
+    }
+    if (ops.size > 1) {
+        throw IllegalArgumentException("Command line has multiple operations: $ops")
+    }
+    val values: MutableMap<String, Any> = mutableMapOf()
+    values[CONNECTOR_OPERATION] = ops.first().name.lowercase()
+    commandLine.optionValue("output")?.let { values[CONNECTOR_OUTPUT_FILE] = it }
+    for ((cliOptionKey, prefix) in
+        mapOf(
+            JavaBaseConstants.ARGS_CONFIG_KEY to CONNECTOR_CONFIG_PREFIX,
+            JavaBaseConstants.ARGS_CATALOG_KEY to CONNECTOR_CATALOG_PREFIX,
+            JavaBaseConstants.ARGS_STATE_KEY to CONNECTOR_STATE_PREFIX,
+        )) {
+        val cliOptionValue = commandLine.optionValue(cliOptionKey) as String?
+        if (cliOptionValue.isNullOrBlank()) {
+            continue
+        }
+        val jsonFile: File = Path.of(cliOptionValue).toFile()
+        if (!jsonFile.exists()) {
+            log.warn { "File '$jsonFile' not found for '$cliOptionKey'." }
```

Comment:
I think we should just crash here? if someone tells us to read a file, and then doesn't provide that file, I'm not sure we can recover gracefully

Reply:
The `.readText()` method below will either take care of that, or will return `""` (not sure which) either way it fails.